crates/pathfinder/src/sequencer.rs

```diff
@@ -292,19 +239,13 @@ impl ClientApi for Client {
         &self,
         block_number: BlockNumberOrTag,
     ) -> Result<reply::Block, SequencerError> {
-        let number = block_number_str(block_number);
-        retry(|| async {
-            let resp = self
-                .inner
-                .get(self.build_query(
-                    &["feeder_gateway", "get_block"],
-                    &[("blockNumber", &number)],
-                ))
-                .send()
-                .await?;
-            parse::<reply::Block>(resp).await
-        })
-        .await
+        self.request()
+            .feeder_gateway()
+            .get_block()
+            .at_block(block_number)
+            .auto_retry()
+            .get()
+            .await
```

Comment:
Example of the finished API usage. Note the lack of surround `retry( || sync { .... })`. I want to move this into `auto_retry` functionally but am failing.